src/api.jl

```diff
@@ -0,0 +1,188 @@
+"""
+    Julja2ParsedTemplate
+
+Represents a parsed template with executable generator code.
+
+## Fields:
+- `generator_code::Expr`: The expression that generates the code for the template.
+"""
+struct Julja2ParsedTemplate
+generator_code::Expr
+end
+
+"""
+    BlockBorders
+
+This structure contains borders of block.
+
+## Fields:
+- `left::String`: The left border string.
+- `left_length::Int`: The length of the left border.
+- `right::String`: The right border string.
+- `right_length::Int`: The length of the right border.
+"""
+mutable struct BlockBorders
+    left::String
+    left_length::Int
+    right::String
+    right_length::Int
+
+    function BlockBorders(left::String, right::String)
+        # Get UInt8 string length because unicode characters may be more than one byte.
+        new(
+            left,
+            length(transcode(UInt8, left)),
+            right,
+            length(transcode(UInt8, right))
+        )
+    end
+end
+
+"""
+    Julja2Template(template::String; var_borders::BlockBorders=BlockBorders("{{", "}}"), 
+    operator_borders::BlockBorders=BlockBorders("{%", "%}"), templates_dir::String="./", 
+    ignore_missing_vars=true, allow_filename_change_dir=false) -> Julja2ParsedTemplate
```

Comment:
```suggestion
    Julja2Template(template::String; kw...) -> Julja2ParsedTemplate
```
In Julia documentation, it is common to remove keyword arguments in the abbreviation `kw` and then describe them in more detail directly inside the docstring (As you did in `Optional parameters`)

Reply:
If you are interested, it will be useful to look [here](https://docs.julialang.org/en/v1/manual/documentation/) to understand the basic rules for writing documentation in Julia